### PR TITLE
RK-6845 - Explorook revisited - settings mutateable and check git auth

### PR DESCRIPTION
**Pull request by amits1995**
settings can be mutated from FE  
new query to check if local git can authenticate with some repository  
git clone automatically tries ssh and https

### CODE REVIEW

File: graphql/schema.graphql [hunk 1]
```diff
@@ -41,6 +41,30 @@ type Query {
...
+  canAuthGitRepos(sources: [GitRepo]!): [CanAuthGitRepoResponse]!
+}
+
+input GitRepo {
+  repoUrl: String!
+}
+
+type CanAuthGitRepoResponse {
+  repoUrl: String!
+  isSuccess: Boolean!
```
Comment (dan505512) [inline graphql/schema.graphql:54]: You can use the response object we made

File: src/git.ts [hunk 2]
```diff
@@ -295,3 +301,34 @@ export function removeGitReposFromStore(folderNames: string[]) {
...
+      resolve(code === 0)
+    })
+  })
+}
+
+export async function canAuthGitRepo(repoUrl: string): Promise<{ querySuccessful: boolean, protocol: GitProtocols }> {
+  const sshFormat = convertUrlToProtocol(repoUrl, GitProtocols.SSH);
+  const httpFormat = convertUrlToProtocol(repoUrl, GitProtocols.HTTPS);
+
+  if (await lsRemote(sshFormat)) {
```
Comment (dan505512) [inline src/git.ts:320]: Looking at our DB I think we should try https first

**Review by dan505512 — CHANGES_REQUESTED** [review]
Please make sure to pull first as I made tiny linting changes

File: src/git.ts (hunk 2 above)
Comment (amits1995) [reply src/git.ts:320]: Our DB has the repo URLs that were used in the CI/CD's so we can't exactly know which protocol the clients use on their local machine.  
Second, if SSH is configured, everything will "just work".  
With HTTP you have multiple choices and managers that hook with the git client that can prompt or provide access.  
For example, if you run git commands (on HTTP remote) from within your VSCode terminal - VSCode injects `GIT_ASKPASS` env var which redirects the git CLI to ask for username/password from VSCode itself, who might prompt the user for credentials.  
Another approach is the git credential.helper

**Review by dan505512 — APPROVED** [review]

File: graphql/schema.graphql (hunk 1 above)
Comment (amits1995) [reply graphql/schema.graphql:54]: Doesn't have `repoUrl`

**Pull request merged**